scipy/stats/tests/test_stats.py

```diff
@@ -3723,123 +3677,158 @@ def do(self, a, b, axis=None, dtype=None):
         assert_equal(x.dtype, dtype)
 
 
-class GeoMeanTestCase:
-    def test_1dlist(self):
+class GeoMeanTestMethod(object):
+    def equal_test(self, array_like, desired, axis=None, dtype=None, decimal=7):
+        # Note this doesn't test when axis is not specified
+        x = stats.gmean(array_like, axis=axis, dtype=dtype)
+        assert_almost_equal(desired, x, decimal=decimal)
+        assert_equal(x.dtype, dtype)
+
+    def approx_test(self, array_like, desired, axis=None, dtype=None, significant=7):
```

Comment:
In this area, I re-used a code from legacy.
Actually `approx_test()` is used in `def test_large_values():`.

In my opinion, 
when the number has a digit(eg. decimal), it is more appropriate.
It is more suitable for  the number is large number.

Actually, two function has different calculation for an equal checking.
```
assert_almost_equal
abs(desired - actual) >= 1.5 * 10.0**(-decimal)

assert_approx_equal
np.abs(sc_desired - sc_actual) >= np.power(10., -(significant-1))
```

Reply:
Sorry for not being clear by saying "Comparing the value for a set number of significant digits" I meant the second formula, I should have just used the actual code. 

I've just re-read the [docs](https://docs.scipy.org/doc/numpy-1.15.0/reference/generated/numpy.testing.assert_approx_equal.html) and both warn:

> It is recommended to use one of assert_allclose, assert_array_almost_equal_nulp or assert_array_max_ulp instead of this function for more consistent floating point comparisons.
 
So both are *probably* not the best choice in this context.

Reply:
No worries, thanks for helping. I think [``assert_allclose`` ](https://docs.scipy.org/doc/numpy-1.15.0/reference/generated/numpy.testing.assert_allclose.html#numpy.testing.assert_allclose) fits best, plus it's already being imported and used elsewhere.